server/core/src/physPath.cpp

```diff
@@ -428,63 +422,38 @@ getSizeInVault( rsComm_t *rsComm, dataObjInfo_t *dataObjInfo ) {
-        // =-=-=-=-=-=-=-
-        // get the resource location for the hier string leaf
-        ret = irods::get_loc_for_hier_string( dataObjInfo->rescHier, location );
-        if ( !ret.ok() ) {
-            irods::log( PASSMSG( "_dataObjChksum - failed in get_loc_for_hier_string", ret ) );
-            return -1;
-        }
+    if (const auto err = irods::get_loc_for_hier_string(dataObjInfo->rescHier, location); !err.ok()) {
+        irods::log(PASSMSG("_dataObjChksum - failed in get_loc_for_hier_string", err));
+        return -1;
+    }
 
-        memset( &fileChksumInp, 0, sizeof( fileChksumInp ) );
-        rstrcpy( fileChksumInp.addr.hostAddr, location.c_str(), NAME_LEN );
-        rstrcpy( fileChksumInp.fileName, dataObjInfo->filePath, MAX_NAME_LEN );
-        rstrcpy( fileChksumInp.rescHier, dataObjInfo->rescHier, MAX_NAME_LEN );
-        rstrcpy( fileChksumInp.objPath,  dataObjInfo->objPath, MAX_NAME_LEN );
-        rstrcpy( fileChksumInp.in_pdmo,  dataObjInfo->in_pdmo, MAX_NAME_LEN );
+    fileChksumInp_t fileChksumInp{};
+    rstrcpy(fileChksumInp.addr.hostAddr, location.c_str(), NAME_LEN);
+    rstrcpy(fileChksumInp.fileName, dataObjInfo->filePath, MAX_NAME_LEN);
+    rstrcpy(fileChksumInp.rescHier, dataObjInfo->rescHier, MAX_NAME_LEN);
+    rstrcpy(fileChksumInp.objPath, dataObjInfo->objPath, MAX_NAME_LEN);
+    rstrcpy(fileChksumInp.in_pdmo, dataObjInfo->in_pdmo, MAX_NAME_LEN);
+    fileChksumInp.dataSize = dataObjInfo->dataSize;
 
-        if ( orig_chksum ) {
-            rstrcpy( fileChksumInp.orig_chksum, orig_chksum, CHKSUM_LEN );
-        }
+    if (const char* orig_chksum = getValByKey(&dataObjInfo->condInput, ORIG_CHKSUM_KW); orig_chksum) {
+        rstrcpy(fileChksumInp.orig_chksum, orig_chksum, CHKSUM_LEN);
+    }
 
-        rodsLog( LOG_DEBUG, "[%s:%d] - performing checksum for [%s] on [%s] at location [%s]",
+    rodsLog(LOG_DEBUG, "[%s:%d] - performing checksum for [%s] on [%s] at location [%s]",
             __FUNCTION__, __LINE__, dataObjInfo->objPath, dataObjInfo->rescHier, dataObjInfo->filePath);
-        status = rsFileChksum( rsComm, &fileChksumInp, chksumStr );
 
-        if ( status == DIRECT_ARCHIVE_ACCESS ) {
-            std::stringstream msg;
-            msg << "Data object: \"";
-            msg << dataObjInfo->filePath;
-            msg << "\" is located in an archive resource. Ignoring its checksum.";
-            irods::log( LOG_DEBUG, msg.str() );
+    const auto ec = rsFileChksum(rsComm, &fileChksumInp, chksumStr);
 
-        }
-        break;
-    default:
-        rodsLog( LOG_NOTICE,
-                 "_dataObjChksum: rescCat type %d is not recognized", category );
-        status = SYS_INVALID_RESC_TYPE;
-        break;
+    if (ec == DIRECT_ARCHIVE_ACCESS) {
+        const auto msg = fmt::format(R"_(Data object: "{}" is located an an archive resource. Ignoring its checksum.)_",
+                                     dataObjInfo->filePath); // TODO Should this be the logical path instead?
```

Comment:
let's consider showing all three pieces of information...
 - resource name (perhaps full hierarchy, at this point it would be helpful to the admin)
 - logical path
 - physical path / OID

Reply:
done.